docs/parallelism.md

```diff
@@ -0,0 +1,44 @@
+# Limiting parallelism
+
+You can restrict the number of workflows being executed at any time using these mechanisms.
+
+## Controller level
+
+You can limit the total number of workflows that can execute at any one time in the [workflow controller ConfigMap](./workflow-controller-configmap.yaml).
+
+```yaml
+data:
+  parallelism: "10"
+```
+
+You can also limit the number of workflows that can execute in a single namespace.
+
+```yaml
+data:
+  namespaceParallelism: "4"
+```
+
+Workflows that are executing but restricted from running more nodes due to the other mechanisms will still count towards the parallelism limits.
```

Comment:
hmmm there is a WIP PR to change this: #12799. if we consider that as "buggy" behavior (as the issue and PR do), then we probably shouldn't document it as it is unintended

Reply:
I don't only mean that mechanism. Suspended workflows will also do this.

I think the original intention would have been not to do what #12799 is trying to fix, because it's simpler. (The current fix there violates parallelism. An aware user could deliberately violate parallelism using it to get their work done.)

I'd like the current documentation to document current behaviour, so I'd like to keep this in.

Reply:
Yea I think there's a few other cases that could all be considered "buggy". The primary difference though, is if it's documented, people won't file bug reports or fix it necessarily -- it's expected behavior. If we link to an existing bug report, or otherwise state that this behavior may be optimized in the future, I think that could be ok as a compromise.

Regarding #12799, please do review there if you have an opinion on the approach! The initial approach was to requeue, which was a lot simpler, the current one is a bit different and I honestly have yet to think through correctness on it